src/Npgsql/CrateDbDatabaseInfo.cs

```diff
@@ -0,0 +1,203 @@
+            // of ArgumentExceptions thrown and log entries created.
+            var unsupportedTypes = mappings.Where(m => !IsPgTypeSupportedByCrateDb(m.Value.PgTypeName)).ToList();
+            foreach (var t in unsupportedTypes)
+            {
+                mappings.Remove(t.Key);
+            }
+
+            // Add mappings specific to CrateDB.
+            foreach (var t in CrateDbSpecificTypeMappings())
+            {
+                if (mappings.ContainsKey(t.PgTypeName))
+                    mappings[t.PgTypeName] = t;
+                else
+                    mappings.Add(t.PgTypeName, t);
+            }
+        }
+
+        /// <summary>
+        /// Returns a boolean value that signals if transactions are supported by the current database.
+        /// </summary>
+        public override bool SupportsTransactions => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the DISCARD command is supported by the current database.
+        /// </summary>
+        public override bool SupportsDiscard => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the DISCARD TEMP subcommand is supported by the current database.
+        /// </summary>
+        public override bool SupportsDiscardTemp => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the DISCARD SEQUENCES subcommand is supported by the current database.
+        /// </summary>
+        public override bool SupportsDiscardSequences => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the CLOSE ALL command is supported by the current database.
+        /// </summary>
+        public override bool SupportsCloseAll => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the UNLISTEN command is supported by the current database.
+        /// </summary>
+        public override bool SupportsUnlisten => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the current database supports range types.
+        /// </summary>
+        public override bool SupportsRangeTypes => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the current database supports advisory lock functions.
+        /// </summary>
+        public override bool SupportsAdvisoryLocks => false;
+        /// <summary>
+        /// Reports whether the backend uses the newer integer timestamp representation. 
+        /// CrateDB used floating point timestamps until version &lt;=3.0 and switched to integer datetimes with version &gt;= 3.1.
+        /// </summary>
+        public override bool HasIntegerDateTimes { get; protected set; } = true;
+
+        static IEnumerable<NpgsqlTypeMapping> CrateDbSpecificTypeMappings()
+        {
+            // Map CrateDB varchar type to the Npgsql TextHandler.
+            yield return new NpgsqlTypeMappingBuilder
```

Comment:
Is this being done because there is no `text` type on Crate, or similar?

Reply:
same as with `timestamptz` :point_up: